safaribooks.py

```diff
@@ -614,7 +614,7 @@ def link_replace(self, link):
             if not self.url_is_absolute(link):
                 if "cover" in link or "images" in link or "graphics" in link or \
                         link[-3:] in ["jpg", "peg", "png", "gif"]:
-                    link = urljoin(self.base_url, link)
+                    link = urljoin(self.base_url + self.book_id + "/", link)
```

Comment:
Aha! This certainly might not work in all cases and I think I found out why.

Consider the two examples:
https://learning.oreilly.com/library/view/spring-into-linux/0131853546/0131853546_ch04lev1sec10.html
points to https://learning.oreilly.com/library/view/spring-into-linux/0131853546/0131853546_ch04lev1sec10_image01.jpeg

However
https://learning.oreilly.com/library/view/rechnerarchitektur-2nd-edition/9783110495362/html/chapter01.xhtml#chapter01_2
the image also points to https://learning.oreilly.com/library/view/rechnerarchitektur-2nd-edition/9783110495362/images/page005_1.jpg

Notice how for the second example the content is hosted one html element deeper `${book-id}/html/${content}` VS just `${book-id}/${content}` which is being used in the first example. The logic that attempts to extract the links thus results in links like `../images/page183_1.jpg` for the second example (the image itself is still hosted under `${book-id}/images` for both cases). This all results in a wrong url when joining.

As a result this is not a fix, is just addresses a symptom of one failing example. The correct fix would be to fix the relative URL extraction that it can actually resolves the URLs correctly relative to the `base_url` (I'll investigate further what is needed for this)

Reply:
Yes your above fix doesn't work on all cases. I opened up another pull-request which should fix all cases here: https://github.com/lorenzodifuccia/safaribooks/pull/218